pandera/checks.py

```diff
@@ -438,22 +451,34 @@ def __repr__(self):
         )
 
 
+T = TypeVar("T")
+
+
 class _CheckMeta(type):  # pragma: no cover
     """Check metaclass."""
 
+    REGISTERED_CUSTOM_CHECKS: Dict[str, Callable] = {}  # noqa
+
     def __getattr__(cls, name: str) -> Any:
         """Prevent attribute errors for registered checks."""
         attr = cls.__dict__.get(name)
         if attr is None:
             raise AttributeError(f"'{cls}' object has no attribute '{name}'")
         return attr
 
+    def __contains__(cls: Type[T], item: Union[T, str]) -> bool:
+        """Allow lookups for registered checks."""
+        if isinstance(item, cls):
+            name = item.name
+            ref = getattr(cls, name, None)
+            return False if ref is None else ref == item
```

Comment:
Custom checks are not tested here. Is it intentional? It seems to be the cause for some tests failing.

I would expect something like

```python
  is_builtin = getattr(cls, item.name, None) is not None
  if is_builtin:
      return True
  is_custom = cls.REGISTERED_CUSTOM_CHECKS.get(item.name, None) is not None
  return is_custom
```

Reply:
Turns out the issue was actually that `ref == item` is a dumb thing to do. Item is callable with the statistics baked in, while ref is the uncurried check. All we need to do is check that the curried name is in the class to pass the tests.

Regarding the previous comment, I would implement the extension API differently. Instead of actually registering the new check onto the class using setattr, I would just add the check into `REGISTERED_CUSTOM_CHECKS` and change the getattr to look in there as well. This may break autocomplete though.

Reply:
Made this change, lemme know what you think.